Tensors/src/main/scala/com/thoughtworks/compute/Tensors.scala

```diff
@@ -1125,14 +1125,52 @@ trait Tensors extends OpenCL {
     }
 
     /** Returns an asynchronous task to read this [[Tensor]] into a [[scala.Array]],
-      * which is linearized in row-majoy order.
+      * which is linearized in row-major order.
       *
       * @group slow
       */
     def flatArray: Future[Array[closure.JvmValue]] = {
       flatBuffer.intransitiveMap(closure.valueType.memory.toArray).run
     }
 
+    /** Recursive function to reshape a flat array into a multi-dimensional one
+      */
+    private[Tensors] def reshapeArray(a:Array[_], b:Array[Int]): Array[_] = {
```

Comment:
Needs a check that the product of the dimensions in `b` equals the length of `a`

Reply:
also renaming those variables to something informative